flow/e2eshared/e2eshared.go

```diff
@@ -65,3 +65,41 @@ func CheckQRecordEquality(t *testing.T, q model.QRecord, other model.QRecord) bo
 
 	return true
 }
+
+// Equals checks if two QRecordBatches are identical.
+func CheckEqualRecordBatches(t *testing.T, q *model.QRecordBatch, other *model.QRecordBatch) bool {
```

Comment:
Maintaining the bool returning version as I intend to move our tests towards `WaitFor` which will want plenty of boolean utility functions